src/sentry/runner/commands/devserver.py

```diff
@@ -227,16 +245,24 @@ def devserver(reload, watchers, workers, experimental_spa, styleguide, prefix, e
     # Make sure that the environment is prepared before honcho takes over
     # This sets all the appropriate uwsgi env vars, etc
     server.prepare_environment()
-    daemons += [("server", ["sentry", "run", "web"])]
+    daemons += [_get_daemon("server")]
 
     if styleguide:
         daemons += [("storybook", ["./bin/yarn", "storybook"])]
 
+    skip_daemons = set(skip_daemons.split(",")) if skip_daemons else set()
+    valid_daemon_names = set(name for (name, cmd) in daemons).union(_DEFAULT_DAEMONS.keys())
```

Comment:
> daemons is guaranteed to be a subset of _DEFAULT_DAEMONS, right?

Actually no, there is the possibility of an `"https"` daemon (which couldn't be extracted without further refactoring because its command is dynamically defined). Also, `"post-process-forwarder"` and `"storybook"` were excluded from `_DEFAULT_DAEMONS` at the time I wrote this, but that's no longer true.

Even so, I agree with your suggestion. The best argument for keeping it here would be the "power user" case where the dev wants to exclude something not foreseen by `_DEFAULT_DAEMONS` (and therefore not documented in the help text). But that's a bad level of compromise: if the need exists, the value should be brought up to `_DEFAULT_DAEMONS` and properly supported. (Or, if that's not worth it, the dev is free to kludge their own solution.)

> You can then move these lines all the way up to the beginning since ideally cli validation should be done first.

Also a big improvement.

I'll change it presently.

Reply:
Thanks for clarifying, and yeah, good compromise. I'm generally of the opinion that complexity should arise only for sufficient use case.